### ISSUE TITLE
Add placeholders to window sidebar: Index

### ISSUE THREAD

**Issue by ggeisler** [ProjectMirador/mirador#1973]
Part of https://github.com/ProjectMirador/mirador/issues/1963

The Index sidebar can show a variety of structures or sequences of canvases. These include a label and may or may not include a canvas thumbnail.

In terms of loading, the [version 2 spec says](https://iiif.io/api/presentation/2.1/#sequence) that the default sequence must be embedded within the manifest and "additional sequences must be referred to from the manifest, not embedded within it...". So depending on whether the user is looking at the default sequence or has selected an alternative sequence, the canvas labels might be available or they might need to be retrieved from an endpoint referenced in the manifest.

So I think both loading and failed states are relevant to the Index sidebar.

### Loading state
Use the skeleton screen pattern to represent the canvas label and (where relevant) the thumbnail:

<img width="315" alt="screen shot 2019-02-26 at 8 42 17 am" src="https://user-images.githubusercontent.com/101482/53425578-89268900-39a2-11e9-9748-6963fed233b4.png">

This is the only type of index sequence we've currently implemented but as we implement others, such as a multi-volume work or the compact view of a list of canvases, the skeleton screen of each type of sequence should map to the elements displayed for that sequence type; in other words, if an index sequence only shows a title, then the skeleton screen for that type of sequence should only show the horizontal rectangle to represent the title, without the thumbnail placeholder.

### Failed state
Both the thumbnail and the canvas label (if it is in an alternate sequence that has to be retrieved from an endpoint) could fail to load.

If the thumbnail fails to load, for example the third one in the mockup below, replace the skeleton screen thumbnail rectangle with a static one:

<img width="315" alt="screen shot 2019-02-26 at 8 42 37 am" src="https://user-images.githubusercontent.com/101482/53425633-a0657680-39a2-11e9-858d-a55ca2513fa5.png">

If the alternate sequence cannot be loaded at all (the referenced file with the sequence cannot be retrieved), display a failure message centered in the sidebar where the first label would otherwise be displayed:

<img width="315" alt="screen shot 2019-02-26 at 8 42 54 am" src="https://user-images.githubusercontent.com/101482/53425688-b8d59100-39a2-11e9-8f5c-80102c953376.png">

**Comment by aeschylus** [comment]
Are skeleton elements intended to have the dimensions of their eventual contents? This is achievable with the manifest metadata, since we know ahead of time the aspect ratio of the resources. If the eventual aspect ratio is not used, then the rendering will cause a janky reflow of the layout when each resource loads. You can experience this on various blogs, for instance, when you're trying to read or scroll, but the images higher up push everything around, making you lose your place. The reflow is more annoying in this case because the user may be trying to click an item when it jumps out of view.

**Comment by jvine** [comment]
Yes, we want to avoid janky reflow as much as possible. If the image size is predictable, the placeholder should reflect it. Text is not necessarily as predictable, but we might choose to truncate/collapse some text areas as we see how it behaves.

**Comment by marlo-longley** [comment]
Audit 2026: not implemented in v4 — no loading or error states exist in the sidebar index components. Tracked in the parent story at #1963.
